gadget/install/install.go

```diff
@@ -207,7 +251,11 @@ func ensureLayoutCompatibility(gadgetLayout *gadget.LaidOutVolume, diskLayout *g
 
 	// Check if all existing device partitions are also in gadget
 	for _, ds := range diskLayout.Structure {
-		if !contains(gadgetLayout.LaidOutStructure, ds) {
+		present, reasonAbsent := contains(gadgetLayout.LaidOutStructure, ds)
+		if !present {
+			if reasonAbsent != "" {
```

Comment:
(nitpick) instead of duplicating the entire fmt.Errorf() we could do something like:
```
var extraInfo string
if reasonAbsent != "" { 
    extraInfo = fmt.Sprintf(": %s", reasonAbsent)
}
return fmt.Errorf("...%s", extraInfo)

Reply:
Or:
```go
diff --git a/gadget/install/install.go b/gadget/install/install.go
index 640bea3acf..91d683dc73 100644
--- a/gadget/install/install.go
+++ b/gadget/install/install.go
@@ -254,9 +254,9 @@ func ensureLayoutCompatibility(gadgetLayout *gadget.LaidOutVolume, diskLayout *g
                present, reasonAbsent := contains(gadgetLayout.LaidOutStructure, ds)
                if !present {
                        if reasonAbsent != "" {
-                               return fmt.Errorf("cannot find disk partition %s (starting at %d) in gadget: %s", ds.Node, ds.StartOffset, reasonAbsent)
+                               reasonAbsent = fmt.Sprintf(": %s", reasonAbsent)
                        }
-                       return fmt.Errorf("cannot find disk partition %s (starting at %d) in gadget", ds.Node, ds.StartOffset)
+                       return fmt.Errorf("cannot find disk partition %s (starting at %d) in gadget%s", ds.Node, ds.StartOffset, reasonAbsent)
                }
        }
 
``` 
but it's a bit ugly :(

Reply:
Applied with a comment.